examples/step-75/step-75.cc

```diff
@@ -0,0 +1,1502 @@
+    for (const auto &cell : dof_handler.active_cell_iterators())
+      if (cell->is_locally_owned())
+        fe_degrees(cell->active_cell_index()) = cell->get_fe().degree;
+
+    Vector<float> subdomain(triangulation.n_active_cells());
+    for (auto &subd : subdomain)
+      subd = triangulation.locally_owned_subdomain();
+
+    DataOut<dim> data_out;
+    data_out.attach_dof_handler(dof_handler);
+    data_out.add_data_vector(locally_relevant_solution, "solution");
+    data_out.add_data_vector(fe_degrees, "fe_degree");
+    data_out.add_data_vector(subdomain, "subdomain");
+    data_out.add_data_vector(estimated_error_per_cell, "error");
+    data_out.add_data_vector(hp_decision_indicators, "hp_indicator");
+    data_out.build_patches(mapping_collection);
+
+    data_out.write_vtu_with_pvtu_record(
+      "./", "solution", cycle, mpi_communicator, 2, 8);
+  }
+
+
+
+  // @sect4{LaplaceProblem::run}
+
+  // The actual run function again looks very familiar to step-40. The only
+  // addition is the bracketed section that preceeds the actual cycle loop.
+  // Here, we will pre-calculate the Legendre transformation matrices. In
+  // general, these will be calculated on the fly via lazy allocation whenever a
+  // certain matrix is needed. For timing purposes however, we would like to
+  // calculate them all at once before the actual time measurement begins. We
+  // will thus designate their calculation to their own scope.
+  template <int dim>
+  void LaplaceProblem<dim>::run()
+  {
+    pcout << "Running with Trilinos on "
+          << Utilities::MPI::n_mpi_processes(mpi_communicator)
+          << " MPI rank(s)..." << std::endl;
+
+    {
+      pcout << "Calculating transformation matrices..." << std::endl;
+      TimerOutput::Scope t(computing_timer, "calculate transformation");
+      legendre->precalculate_all_transformation_matrices();
+    }
+
+    for (unsigned int cycle = 0; cycle < prm.n_cycles; ++cycle)
+      {
+        pcout << "Cycle " << cycle << ':' << std::endl;
+
+        if (cycle == 0)
+          initialize_grid();
+        else
+          adapt_resolution();
+
+        setup_system();
+
+        print_diagnostics();
+
+        solve_system();
+
+        compute_indicators();
+
+        if (Utilities::MPI::n_mpi_processes(mpi_communicator) <= 32)
+          output_results(cycle);
+
+        computing_timer.print_summary();
+        computing_timer.reset();
+
+        pcout << std::endl;
+      }
+  }
+} // namespace Step75
+
+
+
+// @sect4{main()}
+
+// The final function is the <code>main</code> function that will ultimately
+// create and run a LaplceOperator instantiation. Its structure is similar to
+// most other tutorial programs. We define all parameters used in the tutorial
+// here, so this is a good opportunity to justify how we picked them.
+//
+// We provide ranges for levels on which the function space is operating with a
+// reasonable resolution. The minimal polynomial degree is set to 2 as it seems
```

Comment:
> The minimal polynomial degree is set to 2

This should be checked, since I (and also the p-multigrid sequences) am assuming the lowest degree is 1.

Reply:
Since linear elements do not exist in the collection, a new pair with a default value `(1,0)` will be added to the `fe_index_for_degree` map. That's why no error is triggered and the code still works.

The last commit checks if polynomial degrees exist in the `FECollection`. Hence, this tutorial will throw assertions at the moment :(

Reply:
We could also use a `FECollection` with polynomial degrees [1,max], but restrict the hierachy for p-refinement to [2,max].

This way, both precondition and hp-adaptation methods would be happy. What do you think?

Reply:
Looks good!